receiver/datadogreceiver/translator.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogreceiver
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"mime"
+	"net/http"
+	"strings"
+
+	datadogpb "github.com/DataDog/datadog-agent/pkg/trace/exportable/pb"
+	"github.com/tinylib/msgp/msgp"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/model/semconv/v1.6.1"
+)
+
+func toTraces(traces datadogpb.Traces, req *http.Request) pdata.Traces {
```

Comment:
I would test this function thoroughly. I think this is the "meat" of it 🙂

Reply:
Hence the benchmark and functional tests, i just made it quite a bit quicker.